src/main/java/seedu/address/model/student/UniqueStudentList.java

```diff
@@ -241,52 +493,166 @@ create a class from the start time and check to see if it exceeds the startTime
+                    && classToCompare.startTime.compareTo(tr.startTimeRange) > 0) {
+                if (endTimeFromTr.compareTo(classToCompare.startTime) <= 0) {
+                    // trying to fit a slot from the start of the class
+                    newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+                } else if (classToCompare.endTime.plusMinutes(tr.duration).compareTo(tr.endTimeRange) <= 0) {
+                    // fit a class after the end of the class
+                    newClass = new Class(currDate, startTimeFromClass, endTimeFromClass);
+                }
+            } else if (classToCompare.startTime.compareTo(tr.startTimeRange) > 0
+                    && classToCompare.endTime.compareTo(tr.endTimeRange) >= 0) {
+                if (endTimeFromTr.compareTo(classToCompare.startTime) <= 0) {
+                    newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+                }
+            } else if (classToCompare.startTime.compareTo(tr.endTimeRange) >= 0) {
+                newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+            } else {
+                newClass = new Class(currDate.plusDays(1), startTimeFromTr, endTimeFromTr);
+            }
+        } else if (currTime.compareTo(tr.startTimeRange) > 0 && currTime.compareTo(tr.endTimeRange) < 0) {
+            // this means that the currentTime is between the start and end time range
+            if (classToCompare.endTime.compareTo(tr.startTimeRange) <= 0) {
+                if (endTimeFromCurrTime.compareTo(tr.endTimeRange) <= 0) {
+                    newClass = new Class(currDate, startTimeFromCurrTime, endTimeFromCurrTime);
+                }
+            } else if (classToCompare.endTime.compareTo(tr.startTimeRange) > 0
+                    && classToCompare.endTime.compareTo(tr.endTimeRange) < 0
+                    && classToCompare.startTime.compareTo(tr.startTimeRange) <= 0) {
+                if (currTime.compareTo(classToCompare.endTime) <= 0) {
+                    if (endTimeFromClass.compareTo(tr.endTimeRange) <= 0) {
+                        newClass = new Class(currDate, startTimeFromClass, endTimeFromClass);
+                    }
+                } else {
+                    if (endTimeFromCurrTime.compareTo(tr.endTimeRange) <= 0) {
+                        newClass = new Class(currDate, startTimeFromCurrTime, endTimeFromCurrTime);
+                    }
+                }
+            } else if (classToCompare.endTime.compareTo(tr.endTimeRange) < 0
+                    && classToCompare.startTime.compareTo(tr.startTimeRange) > 0) {
+                if (currTime.compareTo(classToCompare.startTime) < 0) {
+                    if (endTimeFromTr.compareTo(classToCompare.startTime) <= 0) {
+                        newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+                    }
+                } else if (currTime.compareTo(classToCompare.startTime) >= 0
```

Comment:
the condition in line 617 can be directly derived from negation of line 613 and can be simplified later.

Reply:
[Missed corner case]
with current time being 2024 and the only lesson is from 2000-2300 (ie including the current time),
when searching "avail 1700-2359 60", the result is "Available date is: 2022-11-06 2300-0000" which ignores the time before the existing lesson.

Reply:
> [Missed corner case] with current time being 2024 and the only lesson is from 2000-2300 (ie including the current time), when searching "avail 1700-2359 60", the result is "Available date is: 2022-11-06 2300-0000" which ignores the time before the existing lesson.

this should be pertaining to a missed check before line 619

Reply:
Resolved!